arbnode/dataposter/storage_test.go

```diff
@@ -3,134 +3,140 @@ package dataposter
 import (
 	"context"
 	"path"
-	"strconv"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/core/rawdb"
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/offchainlabs/nitro/arbnode/dataposter/leveldb"
 	"github.com/offchainlabs/nitro/arbnode/dataposter/redis"
 	"github.com/offchainlabs/nitro/arbnode/dataposter/slice"
+	"github.com/offchainlabs/nitro/arbnode/dataposter/storage"
 	"github.com/offchainlabs/nitro/util/arbmath"
 	"github.com/offchainlabs/nitro/util/redisutil"
 	"github.com/offchainlabs/nitro/util/signature"
 )
 
-func newLevelDBStorage[Item any](t *testing.T) *leveldb.Storage[Item] {
+var ignoreData = cmpopts.IgnoreFields(storage.QueuedTransaction{}, "Data")
```

Comment:
Why are we ignoring data now? Is it a `nil` vs `[]byte{}` issue? If so, I'd like a more specific cutout.

Reply:
Data is already included in signed transaction (FullTx), I'm not sure why do we even have this field in this struct. 

Am I missing something here ?

Reply:
It's just a lot easier to use data in this format, because we can do `types.NewTx(&newTx.Data)` vs creating a new `DynamicFeeTx` there with a copy of every field it needs from the transaction

Reply:
In theory we can, but this is testing internal functionality, and I'm concerned it'd miss an issue where fullTx is correct but somehow Data is not. If the storage backend somehow returned broken data in that way, it would cause problems for the data poster. Is there a specific issue with the Data field being different? If so, ideally we'd have a more targeted resolution there.

Reply:
GetInner() returns an interface instead of the fixed type, which means we'd have to downcast it before mutating the fields. But maybe that makes sense with 4844 coming, because we'll have multiple types of inner tx data.